### PR TITLE
Fix: fix the bug of  vela cli enable addon by localDir on  windows os  

### PR DESCRIPTION
**Pull request by wangyikewxgm**
Signed-off-by: 楚岳 <wangyike.wyk@alibaba-inc.com>

fix several issue

### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/addon/addon.go [hunk 1]
```diff
@@ -195,6 +195,9 @@ func GetPatternFromItem(it Item, r AsyncReader, rootPath string) string {
 	for _, p := range Patterns {
 		if strings.HasPrefix(relativePath, strings.Join([]string{rootPath, p.Value}, "/")) {
 			return p.Value
+		} else if strings.HasPrefix(relativePath, filepath.Join(rootPath, p.Value)) {
+			// for enable addon by load dir, compatible with linux or windows os
+			return p.Value
```
Comment (wonderflow) [inline pkg/addon/addon.go:200]:
```suggestion
		if strings.HasPrefix(relativePath, filepath.Join(rootPath, p.Value)) {
			// for enable addon by load dir, compatible with linux or windows os
			return p.Value
```

File: pkg/addon/addon.go [hunk 2]
```diff
@@ -196,6 +196,10 @@ func GetPatternFromItem(it Item, r AsyncReader, rootPath string) string {
 		if strings.HasPrefix(relativePath, strings.Join([]string{rootPath, p.Value}, "/")) {
 			return p.Value
 		}
+		if strings.HasPrefix(relativePath, filepath.Join(rootPath, p.Value)) {
+			// for enable addon by load dir, compatible with linux or windows os
+			return p.Value
+		}
```
Comment (wonderflow) [inline pkg/addon/addon.go:202]: are they duplicated?

Comment (wangyikewxgm) [reply pkg/addon/addon.go:202]: no, we use this func for all type source(git,oss, local), if type is git oss linux local will meet first if-case, windows will meet second.

Comment (wonderflow) [reply pkg/addon/addon.go:202]: I see

**Review by wonderflow — APPROVED** [review]
let‘’s add some tests

**Pull request merged**